### ISSUE TITLE
mosh client server option makes too many assumptions

### ISSUE THREAD

**Issue by leee** [blinksh/blink#164]
```
blink> mosh leee@athena.dialup.mit.edu --server="athrun mosh_project mosh-server"
Password: # Attempts Kerberos
leee@athena.dialup.mit.edu's password: # Attempts password method
bash: athrun new -s -c 256 -l LC_ALL=en_US.UTF-8 -- mosh_project mosh-server: command not found
Did not find remote IP address
blink> 
```

The mosh client (that is, [Sessions/MoshSession.m](https://github.com/blinksh/blink/blob/25cf8f5f6b8251fa9b469d269997a2e290e5b2e2/Sessions/MoshSession.m)), makes a very dangerous (perhaps even fatal with the right incantation, although I can't think of any at the moment) assumption that the first element in --server option is mosh-server itself.

Server is [set here](https://github.com/blinksh/blink/blob/25cf8f5f6b8251fa9b469d269997a2e290e5b2e2/Sessions/MoshSession.m#L128) passed through [here](https://github.com/blinksh/blink/blob/25cf8f5f6b8251fa9b469d269997a2e290e5b2e2/Sessions/MoshSession.m#L187) which calls [`getMoshServerStringCmd`](https://github.com/blinksh/blink/blob/25cf8f5f6b8251fa9b469d269997a2e290e5b2e2/Sessions/MoshSession.m#L237) that [performs this terrible mutation](https://github.com/blinksh/blink/blob/25cf8f5f6b8251fa9b469d269997a2e290e5b2e2/Sessions/MoshSession.m#L243-L252).

**Comment by carloscabanero** [comment]
Thanks a lot for the review Leee! You are right that I'm only taking the first parameter. Do not think it might be fatal though as the command basically doesn't get validated in any of the mosh launchers of the original project either (mosh.pl or mosh.cc).

**Comment by leee** [comment]
@carloscabanero you're right - I suppose I was overreacting because I was expecting too much from the beginning :^) The handling does indeed appear to be safe (or at least safer than I was making it out to be), and this is still a huge step forward for the mosh client for iOS.

Thank you again for this project!

**Comment by cgull** [comment]
Mosh was developed at MIT, and Athena cluster compatibility is still something we care about, so, yes, not validating that argument is the right thing to do for this specific case and generally too.

**Comment by carloscabanero** [comment]
Thanks @cgull, that explains why. I will fix the bug to send the full --server command then.
